pkg/monitoring/metrics/virt-handler/migrationdomainstats/queue.go

```diff
@@ -0,0 +1,152 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright the KubeVirt Authors.
+ */
+
+package migrationdomainstats
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"k8s.io/client-go/tools/cache"
+
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/log"
+
+	"kubevirt.io/kubevirt/pkg/monitoring/metrics/virt-handler/domainstats"
+	domstatsCollector "kubevirt.io/kubevirt/pkg/monitoring/metrics/virt-handler/domainstats/collector"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/stats"
+)
+
+const (
+	collectionTimeout = 10 * time.Second
+	pollingInterval   = 5 * time.Second
+)
+
+type Result struct {
+	VMI       string
+	Namespace string
+
+	DomainJobInfo stats.DomainJobInfo
+	Timestamp     time.Time
+}
+
+type queue struct {
+	vmiInformer cache.SharedIndexInformer
+
+	vmi     *v1.VirtualMachineInstance
+	results []Result
+
+	isActive  bool
+	mutex     *sync.Mutex
+	collector domstatsCollector.Collector
+}
+
+func newQueue(vmiInformer cache.SharedIndexInformer, vmi *v1.VirtualMachineInstance) *queue {
+	return &queue{
+		vmiInformer: vmiInformer,
+
+		vmi:     vmi,
+		results: make([]Result, 0),
+
+		isActive:  false,
+		mutex:     &sync.Mutex{},
+		collector: domstatsCollector.NewConcurrentCollector(1),
+	}
+}
+
+func (q *queue) startPolling() {
+	q.isActive = true
+
+	ticker := time.NewTicker(pollingInterval)
+	go func() {
+		for range ticker.C {
+			if !q.isActive {
+				ticker.Stop()
+				return
+			}
+			q.collect()
+		}
+	}()
+}
+
+func (q *queue) collect() {
```

Comment:
I would also add some logging, even on start and stop.

Reply:
added